akka-docs-dev/rst/scala/stream-customize.rst

```diff
@@ -129,10 +135,144 @@ Finally, there are two methods available for convenience to complete the stage a
 * ``failStage(exception)`` is equivalent to failing all output ports and cancelling all input ports.
 
 
+In some cases it is inconvenient and error prone to react on the regular state machine events with the
+signal based API described above. For those cases there is a API which allows for a more imperative sequencing
+of actions which will greatly simplify some use cases at the cost of some extra allocations. The difference
+between the two APIs could be described as that the first one is signal driven from the outside, while this API
+is more active and drives its surroundings.
+
+The operations of this part of the :class:``GraphStage`` API are:
+
+* ``emit(out, elem)`` and ``emitMultiple(out, Iterable(elem1, elem2))`` replaces the ``OutHandler`` with a handler that emits
+  one or more elements when there is demand, and then reinstalls the current handlers
+* ``read(in)(andThen)`` and ``readN(in, n)(andThen)`` replaces the handlers that will
+* ``abortEmitting()`` and ``abortReading()`` which will cancel an ongoing emit or read
+
+An example of how this API simplifies a stage can be found below in the second version of the :class:``Duplicator``.
```

Comment:
document that calling setHandler while some of these operations above are still running is inherently dangerous. (like pulling the rug from under emitMultiple)